fuseimg/data/ops/shape_ops.py

```diff
@@ -82,7 +81,214 @@ def __call__(self, sample_dict: NDict, op_id: Optional[str], key: str,
         img = img[slice_idx]
         sample_dict[key] = img
         return sample_dict
+    
+class OpResizeAndPad2D(OpBase):
+    '''
+    Resize and Pad a 2D image
+    '''
+    def __init__(self, number_of_channels : int =1, pad_value :int =0, **kwargs):
+        super().__init__(**kwargs)
+        self.number_of_channels = number_of_channels
+        self.pad_value = pad_value
+    def pad_image(self,inner_image: np.ndarray, padding: Tuple[float, float], resize_to: Tuple[int, int]) -> np.ndarray :
+        """
+        pads image to requested size ,
+        pads both side equally by the same input padding size (left = right = padding[1] , up = down= padding[0] )  ,
+        padding default value is zero or minimum value in normalized target range
+        :param inner_image: image of shape [H, W, C]
+        :param padding: required padding [x,y]
+        :param resize_to: original requested resolution
+        :param normalized_target_range: requested normalized image pixels range
+        :param number_of_channels: number of color channels in the image
+        :return: padded image
+        """
+        inner_image = inner_image.astype('float32')
```

Comment:
Don't force it. Specify that you expect float32 image if it necessary

Reply:
removed